### ISSUE TITLE
`application provision` returns not helpful 500 on client errors

### ISSUE THREAD

**Issue by samedder** [Azure/service-fabric-cli#30]
If an application package is malformed or invalid, the `application provision` API will fail with a 500 response code. This is indicative of a server side error, rather than client and returns no meaningful information about what could be wrong with the application package.

A more detailed response would help reduce debugging times for provision failures.

**Comment by Jose-CF** [comment]
Indeed, I am getting that error but I do believe my application package is fine (maybe no?!) But no way to know it....

**Comment by kkorsakov** [comment]
A workaround may be to print server response body to stderr (as i do locally after fixing msrest lib), usually response is descriptive enough. But this issue really needs attention, sfctl is hard to work with.

**Comment by Jose-CF** [comment]
@kkorsakov the problem is that the server itself only returns 500 and no message.

Fortunately had access to the server and could see in the logs what was wrong...

But really inconvenient that sfctl will not get an error message from the server that would be just displayed... would have saved me and many other people a bunch of hours.

**Comment by kkorsakov** [comment]
SF server always (in my case) returns body along with 500 code. Body contains description in JSON format, but sfctl doesn't print in anywhere. Could you give dump of request/response?

**Comment by Jose-CF** [comment]
Nothing in the body, otherwise I won't be texting here :1234:
